### PR TITLE
refactor(api): make `range` in `nvim_parse_cmd` an array

### PR DESCRIPTION
**Pull request by famiu**
Changes the `range` value in `nvim_parse_cmd` into an array to describe
range information more concisely. Also makes `range` and `count` be
mutually exclusive by making count `-1` when command takes a range
instead of a count. Additionally corrects the behavior of `count` for
built-in commands by making the default count `0`.

### CODE REVIEW

**Comment by muniter** [comment]
Maybe this discussion took place already but why not provide nil instead of -1 which makes it more ergonomic to check in Lua and it's also more idiomatic.

As an example there are many eval functions which one always has too look into the documentation to check if it retuns -1 or 0 and in which cases, etc.

No strong opinion, just wondering.

**Review by justinmk — APPROVED** [review]

File: src/nvim/api/vimscript.c [hunk 1]
```diff
@@ -849,15 +848,24 @@ Dictionary nvim_parse_cmd(String str, Dictionary opts, Error *err)
...
+    ADD(range, INTEGER_OBJ(ea.line2));
+    PUT(result, "range", ARRAY_OBJ(range));;
+  } else {
+    PUT(result, "range", ARRAY_OBJ(ARRAY_DICT_INIT));
+  }
 
   if (ea.argt & EX_COUNT) {
-    if (ea.addr_count > 0 || cmd == NULL) {
+    if (ea.addr_count > 0) {
       PUT(result, "count", INTEGER_OBJ(ea.line2));
```
Comment (justinmk) [inline src/nvim/api/vimscript.c:864]: should this be `ea.addr_count` ?

Comment (famiu) [reply src/nvim/api/vimscript.c:864]: Are you asking if `ea.addr_count` is the variable that needs to be checked? Then yes.

File: src/nvim/api/vimscript.c [hunk 2]
```diff
@@ -747,13 +747,12 @@ Dictionary nvim_parse_expression(String expr, String flags, Boolean highlight, E
...
-///                           |<line1>|
-///         - line2: (number) Final line of command |<range>|. -1 if command cannot take a range.
-///                           |<line2>|
+///         - range: (array) Command <range>. Can have 0-2 elements depending on how many items the
+///                          range contains. Has no elements if command doesn't accept a range or if
+///                          no range was specified, one element if only a single range item was
+///                          specified and two elements if both range items were specified.
 ///         - count: (number) Any |<count>| that was supplied to the command. -1 if command cannot
-///                           take a count.
+///                           take a count. Mutually exclusive from "range".
```
Comment (justinmk) [inline src/nvim/api/vimscript.c:755]:
```suggestion
///                           take a count. Mutually exclusive with "range".
```

**Pull request merged**